apps/meteor/client/views/room/Header/Omnichannel/QuickActions/hooks/useQuickActions.tsx

```diff
@@ -309,8 +309,8 @@ export const useQuickActions = (): {
 	const canSendTranscriptPDF = usePermission('request-pdf-transcript');
 	const canCloseRoom = usePermission('close-livechat-room');
 	const canCloseOthersRoom = usePermission('close-others-livechat-room');
-	const canPlaceChatOnHold = Boolean(!room.onHold && room.u && !(room as any).lastMessage?.token && manualOnHoldAllowed);
-
+	const allowManualOnHoldUponAgentEngagement = useSetting('Livechat_allow_manual_on_hold_upon_agent_engagement_only');
+	const canPlaceChatOnHold = !room.onHold && room.u && manualOnHoldAllowed && (!allowManualOnHoldUponAgentEngagement || !(room as any).lastMessage?.token);
```

Comment:
```suggestion
	const canPlaceChatOnHold = Boolean(!room.onHold && room.u && manualOnHoldAllowed && (!allowManualOnHoldUponAgentEngagement || !(room as any).lastMessage?.token));
```
TypeScript error: undefined is not assignable to boolean.